src/redux/slices/keyring.ts

```diff
@@ -252,6 +252,37 @@ export const validatePem = createAsyncThunk(
   }
 );
 
+export const getPemFile = createAsyncThunk(
+  'keyring/getPemFile',
+  async (
+    {
+      walletId,
+      onSuccess,
+      onFailure,
+      onFinish,
+    }: {
+      walletId: string;
+      onSuccess: (content: string) => void;
+      onFailure?: () => void;
+      onFinish?: () => void;
+    },
+    { rejectWithValue }
+  ) => {
+    try {
+      const instance = KeyRing.getInstance();
+      const response = await instance?.getPemFile(walletId);
+      if (response) {
+        await onSuccess(response);
+      } else {
+        onFailure?.();
+      }
+      onFinish?.();
+    } catch (e: any) {
+      return rejectWithValue({ error: e.message });
```

Comment:
catch should call onFinish too?
also, `rejectWithValue(e.message);`

Reply:
Added!